PreFreeSurfer/scripts/BiasFieldCorrection_T1wOnly.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+set -e
+source ${HCPPIPEDIR}/global/scripts/debug.shlib # Debugging functions
+
+# Requirements for this script
+#  installed versions of: FSL (version 5.0.6), caret7 (a.k.a. Connectome Workbench) (version 1.0)
+#  environment: FSLDIR  CARET7DIR
+
+################################################ SUPPORT FUNCTIONS ##################################################
+
+Usage() {
+  echo "`basename $0`: Tool for bias field correction based on T1w image only"
+  echo " "
+  echo "Usage: `basename $0` --workingdir=<working directory> --T1im=<input T1 image> [--oT1im=<output T1 image>] [-oT1brain=<output T1 brain>] [--bfsigma=<input T1 image>]"
+}
+
+# function for parsing options
+getopt1() {
+    sopt="$1"
+    shift 1
+    for fn in $@ ; do
+      	if [ `echo $fn | grep -- "^${sopt}=" | wc -w` -gt 0 ] ; then
+      	    echo $fn | sed "s/^${sopt}=//"
+      	    return 0
+      	fi
+    done
+}
+
+defaultopt() {
+    echo $1
+}
+
+################################################### OUTPUT FILES #####################################################
+
+# Output images and files (in $WD):
+# T1.nii.gz                         T1_fast_bias_vol2.nii.gz          T1_initfast2_brain_mask2.nii.gz
+# T1_biascorr.nii.gz                T1_fast_bias_vol32.nii.gz         T1_initfast2_maskedrestore.nii.gz
+# T1_biascorr_brain.nii.gz          T1_fast_restore.nii.gz            T1_initfast2_restore.nii.gz
+# T1_biascorr_brain_mask.nii.gz     T1_fast_seg.nii.gz                lesionmask.nii.gz
+# T1_fast_bias.nii.gz               T1_fast_totbias.nii.gz            lesionmaskinv.nii.gz
+# T1_fast_bias_idxmask.nii.gz       T1_initfast2_brain.nii.gz         log.txt
+# T1_fast_bias_init.nii.gz          T1_initfast2_brain_mask.nii.gz
+
+################################################## OPTION PARSING #####################################################
+
+# Just give usage if no arguments specified
+if [ $# -eq 0 ] ; then Usage; exit 0; fi
+# check for correct options
+if [ $# -lt 2 ] ; then Usage; exit 1; fi
+
+# parse arguments
+WD=`getopt1 "--workingdir" $@`  
+T1wImage=`getopt1 "--T1im" $@`  
+T1wBrain=`getopt1 "--T1brain" $@`  
+oBias=`getopt1 "--obias" $@`  
+oT1wImage=`getopt1 "--oT1im" $@`  
+oT1wBrain=`getopt1 "--oT1brain" $@`  
+BiasFieldSmoothingSigma=`getopt1 "--bfsigma" $@`
+
+# default parameters
+BiasFieldSmoothingSigma=`defaultopt $BiasFieldSmoothingSigma 20` 
```

Comment:
Add comment on why the default sigma is different from that in `BiasFieldCorrection_sqrtT1wXT1w.sh`

Reply:
Would
```
# A default value of 20 for bias smoothing sigma is the recommended default by FSL
``` 
suffice?

Reply:
For me, yes, although I wonder if @glasserm will have any concerns about the very different default value.

Reply:
I'm happy to set it to what @glasserm believes is the right value. My assumption was that the difference is due to using a single rather than two modalities.

Reply:
A sigma of 20 is pretty huge and probably wouldn't be effective for e.g. a 32 channel head coil without pre-scan normalize.  On the other hand we get away with a sigma of 5 with the sqrt(T1w*T2w) approach because it has very little grey/white contrast and the CSF is masked out.  I am fine with the legacy mode using the FSL default.